projects/hslayers/src/components/add-data/vector/add-data-vector.component.ts

```diff
@@ -82,10 +98,41 @@ export class HsAddDataVectorComponent {
     }
   }
 
+  setUploadType(type: string): void {
+    this.uploadType = type;
+    if (type == 'existing') {
+      this.vectorLayers = this.hsLayerManagerService.data.layers.filter(
+        (layer) => {
+          return this.hsLayerUtilsService.isLayerVectorLayer(layer.layer);
+        }
+      );
+    }
+  }
+
   /**
    * Handler for adding non-wms service, file in template.
    */
   async add() {
+    this.uploadType == 'new'
+      ? await this.addNewLayer()
+      : await this.updateExistingLayer();
+
+    this.hsLayoutService.setMainPanel('layermanager');
+    this.setToDefault();
+  }
+
+  async updateExistingLayer(): Promise<void> {
+    let features = this.features.length > 0 ? this.features : [];
+    if (this.dataType != 'geojson') {
+      const kml = await this.hsAddDataVectorService.convertUploadedData(
+        this.vectorFileInput.nativeElement.files[0]
+      );
+      features = kml.features; //proper typing will get rid of this
+    }
+    this.sourceLayer.getSource().addFeatures(features);
```

Comment:
Could you check if this is clustered layer and add to getSource().getSource() in that case?